core/builder/src/main/java/io/quarkus/builder/BuildChainBuilder.java

```diff
@@ -64,6 +66,11 @@
      * @return the builder for the build step
      */
     public BuildStepBuilder addBuildStep(BuildStep buildStep) {
+        final String id = buildStep.toString();
```

Comment:
I'm a bit concerned about using these string values for ordering purposes since they're pretty arbitrary.  I wonder if you considered any other approaches to determining an ordinal value for each step?

Reply:
This worked for me in practice, because most of the build steps happen to have a quite well defined `toString()`. I would not mind modelling the string sort key properly, but you perhaps have a better idea?

Reply:
I began work on this about a year ago but only got a couple ideas in code.  The first was this: https://github.com/dmlloyd/quarkus/commit/build-ordinals which defines a class that represents an overall "path" style sequence.  The second I can't find which sets the ordinal value for a `StepInfo` based on its dependency path.  The dependency with the lowest ordinal is the "parent" of a given step's ordinal, and a step has no parent if it has no dependencies.  The ordinals are assigned sequentially and (I think) deterministically (because the extension load process is sequental, so for a given build, it should run in the same order each time).

Reply:
`buildStep.toString()` is not used for ordering anymore in aacdcf7